tests/ui_tools/test_boxes.py

```diff
@@ -815,6 +815,66 @@ def test_generic_autocomplete_user_mentions(
                 },
                 {"pinned": ["Secret stream", "Stream 1"]},
             ),
+            # With 'Secret stream' muted.
+            (
+                "#Stream",
+                {
+                    0: "#**Stream 1**",
+                    1: "#**Stream 2**",
+                    2: "#**Some general stream**",
+                    3: "#**Web public stream**",
+                    4: "#**Secret stream**",
+                },
+                {"muted": {99}},
+            ),
+            # With 'Stream 1' and 'Secret stream' muted.
+            (
+                "#Stream",
+                {
+                    0: "#**Stream 2**",
+                    1: "#**Some general stream**",
+                    2: "#**Web public stream**",
+                    3: "#**Stream 1**",
+                    4: "#**Secret stream**",
+                },
+                {"muted": {99, 1}},
+            ),
+            # With 'Stream 1' and 'Secret stream' pinned, 'Secret stream' muted.
+            (
+                "#Stream",
+                {
+                    0: "#**Stream 1**",
+                    1: "#**Stream 2**",
+                    2: "#**Some general stream**",
+                    3: "#**Web public stream**",
+                    4: "#**Secret stream**",
+                },
+                {"pinned": ["Secret stream", "Stream 1"], "muted": {99}},
+            ),
+            # With 'Stream 1' and 'Secret stream' pinned, 'Some general stream' and 'Stream 2' muted.
+            (
+                "#Stream",
+                {
+                    0: "#**Stream 1**",
+                    1: "#**Secret stream**",
+                    2: "#**Web public stream**",
+                    3: "#**Stream 2**",
+                    4: "#**Some general stream**",
+                },
+                {"pinned": ["Secret stream", "Stream 1"], "muted": {1000, 2}},
+            ),
+            # With 'Stream 1' and 'Secret stream' pinned, 'Secret stream' and 'Stream 2' muted.
+            (
+                "#Stream",
+                {
+                    0: "#**Stream 1**",
+                    1: "#**Some general stream**",
+                    2: "#**Web public stream**",
+                    3: "#**Secret stream**",
+                    4: "#**Stream 2**",
+                },
+                {"pinned": ["Secret stream", "Stream 1"], "muted": {99, 2}},
+            ),
```

Comment:
In this commit where you add the test cases, you're not connecting the muted elements to the muted_streams data.

Reply:
Yes, it's there now, but it had disappeared in the previous iteration it seemed.